### ISSUE TITLE
dash not replaced following formatted text

### ISSUE THREAD

**Issue by chrylis** [asciidoctor/asciidoctor#1578]
I have a word in italics that should be followed by a dash:

```
_italicized_--but nevertheless...
```

Asciidoctor does not replace the two hyphens with a dash as usual. Removing the underscores makes the substitution work, but changing them to asterisks (bold) reintroduces the issue. It seems that the replacement-detection phase gets confused by adjacent non-text characters that will be elided into formatting.

**Comment by mojavelinux** [comment]
> replacement-detection phase gets confused by adjacent non-text characters that will be elided into formatting.

That's correct. This is just a limitation in Asciidoctor right now. It's not really possible to fix this generally until we switch to a formal parser for inline syntax (see #61). Until then, I recommend using an attribute instead of the double hyphen.

```
:emdash: &#8212;&#8203;

_italicized_{emdash}but nevertheless
```

I know it's not as elegant, but it is dependable.

**Comment by chrylis** [comment]
That's the workaround I'm already using for dashes after internal (aside) exclamation points, so I'll just keep that in mind.

Regarding the parser, at a quick take it looks like Asciidoctor is context-free; is that correct?

**Comment by mojavelinux** [comment]
AsciiDoc is mostly context free, particularly for inline markup. Where it is not, we will address those cases as we work on the parser. Likely any changes that need to be made will be good for the language anyway (making it easier for tools).

**Comment by mojavelinux** [comment]
This one really bites me all the time. We need to find a solution for this.

**Comment by mojavelinux** [comment]
A dash also doesn't work immediately following a period. However, I think we should consider supporting this case.

**Comment by gregturn** [comment]
Upvote.

I'm seeing this all over my manuscript. Thanks for the work-around.

**Comment by gregturn** [comment]
Hmm. The work around doesn't QUITE work.

I forced an em-dash thanks to the macro...but it stopped the backticks from causing a monospaced formatting to occur.

<img width="532" alt="Screen Shot 2020-05-11 at 9 53 28 AM" src="https://user-images.githubusercontent.com/311040/81576031-58472f80-936d-11ea-8642-f72dc82a65ed.png">
